src/main/java/com/iexec/core/task/update/TaskUpdateManager.java

```diff
@@ -180,9 +163,14 @@ Task updateTaskStatusAndSave(Task task, TaskStatus newStatus) {
     Task updateTaskStatusAndSave(Task task, TaskStatus newStatus, ChainReceipt chainReceipt) {
         TaskStatus currentStatus = task.getCurrentStatus();
         task.changeStatus(newStatus, chainReceipt);
-        Task savedTask = taskRepository.save(task);
-        log.info("UpdateTaskStatus suceeded [chainTaskId:{}, currentStatus:{}, newStatus:{}]", task.getChainTaskId(), currentStatus, newStatus);
-        return savedTask;
+        Optional<Task> savedTask = taskService.updateTask(task);
+        if (savedTask.isPresent()) {
+            log.info("UpdateTaskStatus succeeded [chainTaskId:{}, currentStatus:{}, newStatus:{}]", task.getChainTaskId(), currentStatus, newStatus);
+            return savedTask.get();
+        } else {
+            log.warn("UpdateTaskStatus failed [chainTaskId:{}, currentStatus:{}, wishedStatus:{}]", task.getChainTaskId(), currentStatus, newStatus);
+            return null;
```

Comment:
This value is never checked in calling methods.
If the update fails, there will only be a warn in the log and the scheduler will never be aware of it.
What should the behavior be when calling `updateTaskStatusAndSave` in succession if one of them fails ?
example
```
updateTaskStatusAndSave(task, FINALIZE_FAILED);
updateTaskStatusAndSave(task, FAILED);
```

Reply:
Not sure whether it's a good idea as it could lead to infinite loop if `taskService.updateTask(...)` always fails - which would be a serious issue itself.

BTW, I've just checked the `repository#save` method JavaDoc and it states the following:
> @return the saved entity; will never be {@literal null}.

So we should never encounter this case if the task exists. To be clear, I'll add a more explicit comment to the `updateTaskStatusAndSave` method.